mvc-login/src/main/java/com/auth0/example/SecurityConfig.java

```diff
@@ -1,41 +1,45 @@
 package com.auth0.example;
 
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.method.configuration.EnableMethodSecurity;
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
-import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
 import org.springframework.security.web.SecurityFilterChain;
-import org.springframework.security.web.util.matcher.AntPathRequestMatcher;
+import org.springframework.security.web.authentication.logout.LogoutHandler;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
 
-@EnableWebSecurity
-public class SecurityConfig {
+import java.io.IOException;
 
-    private final LogoutHandler logoutHandler;
+import static org.springframework.security.config.Customizer.withDefaults;
 
-    public SecurityConfig(LogoutHandler logoutHandler) {
-        this.logoutHandler = logoutHandler;
-    }
+@Configuration
+@EnableMethodSecurity(securedEnabled = true)
+public class SecurityConfig {
+
+    @Value("${okta.oauth2.issuer}")
+    private String issuer;
+    @Value("${okta.oauth2.client-id}")
+    private String clientId;
 
     @Bean
-    public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
-        http.authorizeRequests()
-                // allow all users to access the home pages and the static images directory
-                .mvcMatchers("/", "/images/**").permitAll()
-                // all other requests must be authenticated
-                .anyRequest().authenticated()
-                .and().oauth2Login()
-                .and().logout()
-                // handle logout requests at /logout path
-                .logoutRequestMatcher(new AntPathRequestMatcher("/logout"))
-                // customize logout handler to log out of Auth0
-                .addLogoutHandler(logoutHandler);
-        return http.build();
+    public SecurityFilterChain configure(HttpSecurity http) throws Exception {
+        return http.authorizeHttpRequests((requests) -> requests
+                .requestMatchers("/", "/images/**").permitAll()
+                    .anyRequest().authenticated())
+                .oauth2Login(withDefaults())
+                .logout(logout -> logout.addLogoutHandler(logoutHandler()))
+                .build();
     }
 
-// If using HS256, create a Bean to specify the HS256 should be used. By default, RS256 will be used.
-//    @Bean
-//    public JwtDecoderFactory<ClientRegistration> idTokenDecoderFactory() {
-//        OidcIdTokenDecoderFactory idTokenDecoderFactory = new OidcIdTokenDecoderFactory();
-//        idTokenDecoderFactory.setJwsAlgorithmResolver(clientRegistration -> MacAlgorithm.HS256);
-//        return idTokenDecoderFactory;
-//    }
+    private LogoutHandler logoutHandler() {
```

Comment:
Ideally we'd use Spring's support for RP-initiated logout, but Auth0 does not currently publish the `end_session_endpoint` URL in the discovery metadata, so we redirect to the Auth0 logout endpoint.

Reply:
I was able to get this endpoint enabled in one of my tenants. Do you know when it'll be enabled by default for everyone?

Reply:
I don't; I'll try and see if I can find out any additional info. I assume when the endpoint was enabled the RP-initiated logout worked?

Reply:
Yes, the RP-initiated logout worked as expected.